qiling/loader/pe.py

```diff
@@ -325,6 +332,11 @@ def load(self):
         self.ql.nprint("[+] Initiate stack address at 0x%x " % self.stack_address)
         self.ql.mem.map(self.stack_address, self.stack_size, info="[stack]")
 
+        self.EXIT_TRAP_ADDR = self.ql.os.heap.alloc(8)
+        # write nop to THREAD_RET_ADDR
+        self.ql.mem.write(self.EXIT_TRAP_ADDR, b"\x90"*8)
```

Comment:
i think you dont need to write anything to the trap address, as callback make emulation stop without exec anything there.

Reply:
Indeed.
The address should be valid (since the `ret` is evaluated), but the contents of the address do not matter.